app/src/main/java/org/wikipedia/language/LangLinksActivity.kt

```diff
@@ -0,0 +1,367 @@
+    private var actionMode: ActionMode? = null
+    private var siteInfoList: List<SiteMatrix.SiteInfo>? = null
+
+    private val entriesByAppLanguages: List<PageTitle>
+        get() {
+            val list = mutableListOf<PageTitle>()
+            for (entry in languageEntries) {
+                if (entry.wikiSite.languageCode() == AppLanguageLookUpTable.NORWEGIAN_LEGACY_LANGUAGE_CODE &&
+                        app.language().appLanguageCodes.contains(AppLanguageLookUpTable.NORWEGIAN_BOKMAL_LANGUAGE_CODE) ||
+                        app.language().appLanguageCodes.contains(entry.wikiSite.languageCode())) {
+                    list.add(entry)
+                }
+            }
+            return list
+        }
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLanglinksBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        title = intent.getParcelableExtra(EXTRA_PAGETITLE)!!
+        binding.langlinkEmptyView.visibility = View.GONE
+        binding.langlinksLoadProgress.visibility = View.VISIBLE
+        fetchLangLinks()
+
+        binding.langlinksError.backClickListener = View.OnClickListener { onBackPressed() }
+        binding.langlinksError.retryClickListener = View.OnClickListener {
+            ViewAnimations.crossFade(binding.langlinksError, binding.langlinksLoadProgress)
+            fetchLangLinks()
+        }
+    }
+
+    public override fun onDestroy() {
+        disposables.clear()
+        super.onDestroy()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.menu_languages_list, menu)
+        val searchIcon = menu.getItem(0)
+        searchIcon.isVisible = languageEntries.isNotEmpty()
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.menu_search_language -> {
+                if (actionMode == null) {
+                    actionMode = startSupportActionMode(LanguageSearchCallback())
+                }
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onBackPressed() {
+        DeviceUtil.hideSoftKeyboard(this)
+        super.onBackPressed()
+    }
+
+    private inner class LanguageSearchCallback : SearchActionModeCallback() {
+        private val langLinksAdapter = binding.langlinksRecycler.adapter as LangLinksAdapter
+
+        override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
+            actionMode = mode
+            return super.onCreateActionMode(mode, menu)
+        }
+
+        override fun onQueryChange(s: String) {
+            currentSearchQuery = s.trim()
+            langLinksAdapter.setFilterText(currentSearchQuery!!)
+            if (binding.langlinksRecycler.adapter!!.itemCount == 0) {
```

Comment:
Can we use `binding.langlinksRecycler.adapter?.itemCount` for the null check on the adapter?